src/otx/backend/openvino/models/base.py

```diff
@@ -0,0 +1,346 @@
+        if "preset" in initial_ptq_config:
+            initial_ptq_config["preset"] = QuantizationPreset(initial_ptq_config["preset"])
+            argparser.add_argument("--preset", type=QuantizationPreset)
+        if "model_type" in initial_ptq_config:
+            initial_ptq_config["model_type"] = ModelType(initial_ptq_config["model_type"])
+            argparser.add_argument("--model_type", type=ModelType)
+        if "ignored_scope" in initial_ptq_config:
+            argparser.add_class_arguments(IgnoredScope, "ignored_scope", as_positional=True)
+
+        initial_ptq_config = argparser.parse_object(initial_ptq_config)
+
+        return argparser.instantiate_classes(initial_ptq_config).as_dict()
+
+    def prepare_metric_inputs(
+        self,
+        preds: OTXPredBatch,
+        inputs: OTXDataBatch,
+    ) -> MetricInput:
+        """Convert prediction and input entities to a format suitable for metric computation.
+
+        Args:
+            preds (OTXPredBatch): The predicted batch entity containing predicted labels.
+            inputs (OTXDataBatch): The input batch entity containing ground truth labels.
+
+        Returns:
+            MetricInput: A dictionary contains 'preds' and 'target' keys
+            corresponding to the predicted and target labels for metric evaluation.
+        """
+        raise NotImplementedError
+
+    def compute_metrics(self, meter: Metric) -> dict:
+        """Compute metrics using the provided torchvision Metric."""
+        return self._compute_metrics(meter)
+
+    def _compute_metrics(self, meter: Metric, **compute_kwargs) -> dict:
+        sig = inspect.signature(meter.compute)
+        filtered_kwargs = {key: value for key, value in compute_kwargs.items() if key in sig.parameters}
+        if removed_kwargs := set(compute_kwargs.keys()).difference(filtered_kwargs.keys()):
+            msg = f"These keyword arguments are removed since they are not in the function signature: {removed_kwargs}"
+            logger.debug(msg)
+
+        results: dict[str, Tensor] = meter.compute(**filtered_kwargs)
+
+        if not isinstance(results, dict):
+            raise TypeError(results)
+
+        if not results:
+            msg = f"{meter} has no data to compute metric or there is an error computing metric"
+            raise RuntimeError(msg)
+        return results
+
+    @property
+    def model_adapter_parameters(self) -> dict:
+        """Model parameters for export."""
+        return {}
+
+    @property
+    def label_info(self) -> LabelInfo:
+        """Get this model label information."""
+        return self._label_info
+
+    @property
+    def task(self) -> OTXTaskType | None:
+        """Get task of the model."""
+        return self._task
+
+    def _create_label_info_from_ov_ir(self) -> LabelInfo:
+        ov_model = self.model.get_model()
+
+        if ov_model.has_rt_info(["model_info", "label_info"]):
```

Comment:
sorry not related to this pr
but I spent some time to guess what `rt` would mean :)
What does it stand for

Reply:
I guess, it stands for "real_time", this name is used by OV.